pkg/prom/wal/wal.go

```diff
@@ -566,78 +565,54 @@ type appender struct {
 }
 
 func (a *appender) Append(ref uint64, l labels.Labels, t int64, v float64) (uint64, error) {
-	if ref == 0 {
-		return a.Add(l, t, v)
-	}
-	return ref, a.AddFast(ref, t, v)
-}
-
-func (a *appender) Add(l labels.Labels, t int64, v float64) (uint64, error) {
-	hash := l.Hash()
-	series := a.w.series.getByHash(hash, l)
-	if series != nil {
-		return series.ref, a.AddFast(series.ref, t, v)
-	}
-
```

Comment:
tsdb.Head in 2.26 doesn't do a getByHash first 🤷‍♂️ Leaving it out of the new impl.